lib/assessment.js

```diff
@@ -8,15 +8,39 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
+var min = function min(array, callback, initialValue) {
+
+  var previousValue;
+
+    array = array.slice();
+    previousValue = array.shift();
+  }
+
+  array.forEach(function(currentValue, index, array) {
+      previousValue = callback(previousValue, currentValue, index, array);
+  });
+
+  return previousValue;
 
 };
 
+var number = function number(previousValue, currentValue, index, array) {
+  return Math.min(array);
+};
+
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
 var sum = function sum() {
 
+
+
+// from MDN:
+// var total = [0, 1, 2, 3].reduce(function(a, b) {
+//   return a + b;
+// });
+// // total == 6
```

Comment:
That's exactly right, actually - the callback in there will cause all of the values to get added up.